assets/javascripts/discourse-assign/initializers/extend-for-assigns.js.es6

```diff
@@ -312,15 +327,58 @@ function initialize(api) {
   api.addKeyboardShortcut("g a", "", { path: "/my/activity/assigned" });
 }
 
+const REGEXP_USERNAME_PREFIX = /^(assigned:)/gi;
+
 export default {
   name: "extend-for-assign",
   initialize(container) {
     const siteSettings = container.lookup("site-settings:main");
     if (!siteSettings.assign_enabled) {
       return;
     }
+    const currentUser = container.lookup("current-user:main");
+    if (currentUser.can_assign) {
+      SearchAdvancedOptions.reopen({
+        _init() {
+          this._super();
+
+          this.set("searchedTerms.assigned", "");
+        },
+
+        @observes("searchedTerms.assigned")
+        updateSearchTermForAssignedUsername() {
+          const match = this.filterBlocks(REGEXP_USERNAME_PREFIX);
+          const userFilter = this.get("searchedTerms.assigned");
+          let searchTerm = this.searchTerm || "";
+          let keyword = "assigned";
+          if (userFilter && userFilter.length !== 0) {
+            if (match.length !== 0) {
+              searchTerm = searchTerm.replace(
+                match[0],
+                `${keyword}:${userFilter}`
+              );
+            } else {
+              searchTerm += ` ${keyword}:${userFilter}`;
+            }
+
+            this.set("searchTerm", searchTerm.trim());
+          } else if (match.length !== 0) {
+            searchTerm = searchTerm.replace(match[0], "");
+            this.set("searchTerm", searchTerm.trim());
+          }
+        },
+
+        _update() {
+          this._super(...arguments);
+          this.setSearchedTermValue(
+            "searchedTerms.assigned",
+            REGEXP_USERNAME_PREFIX
+          );
+        },
+      });
+    }
 
-    withPluginApi("0.8.11", (api) => initialize(api, container));
+    withPluginApi("0.11.0", (api) => initialize(api, container));
     withPluginApi("0.8.28", (api) =>
       registerTopicFooterButtons(api, container)
     );
```

Comment:
Why not merge the 2 calls to `withPluginApi`?

```suggestion
    withPluginApi("0.11.0", (api) => { 
      initialize(api, container);
      registerTopicFooterButtons(api, container);
    });
```

Reply:
We need to use different API versions. The new search dropdown APIs are only in 0.11.0, but we want `registerTopicFooterButtons` to continue working on the stable version of core.